plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,207 @@
+  isUserEntity,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import { Progress } from '@backstage/core-components';
+import {
+  EntityCardActions,
+  UserCardActions,
+  GroupCardActions,
+} from './CardActionComponents';
+import { EntityNotFoundCard } from './EntityNotFoundCard';
+import { debounce } from 'lodash';
+
+/**
+ * Properties for an entity popover on hover of a component.
+ *
+ * @public
+ */
+export type EntityPeekAheadPopoverProps = PropsWithChildren<{
+  entityRef: string;
+  delayTime?: number;
+}>;
+
+const useStyles = makeStyles(() => {
+  return {
+    trigger: {
+      display: 'inline-block',
+    },
+    popoverPaper: {
+      width: '30em',
+    },
+    descriptionTypography: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      display: '-webkit-box',
+      WebkitLineClamp: 2,
+      WebkitBoxOrient: 'vertical',
+    },
+  };
+});
+
+const maxTagChips = 4;
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+export const EntityPeekAheadPopover = (props: EntityPeekAheadPopoverProps) => {
+  const { entityRef, children, delayTime = 500 } = props;
+
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const popupState = usePopupState({
+    variant: 'popover',
+    popupId: 'entity-peek-ahead',
+  });
+  const compoundEntityRef = parseEntityRef(entityRef);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const debouncedHandleMouseEnter = debounce(
+    () => setIsHovered(true),
+    delayTime,
+  );
+
+  const [{ loading, error, value: entity }, load] = useAsyncFn(async () => {
+    const catalogApi = apiHolder.get(catalogApiRef);
+    if (catalogApi) {
+      const retrievedEntity = await catalogApi.getEntityByRef(
+        compoundEntityRef,
+      );
+      if (!retrievedEntity) {
+        throw new Error(`${compoundEntityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [apiHolder, compoundEntityRef]);
+
+  const handleOnMouseLeave = () => {
+    setIsHovered(false);
+    debouncedHandleMouseEnter.cancel();
+  };
+
+  useEffect(() => {
+    if (popupState.isOpen && !entity && !error && !loading) {
+      load();
+    }
+  }, [popupState.isOpen, load, entity, error, loading]);
+
+  return (
+    <>
+      {' '}
+      <span
+        onMouseEnter={debouncedHandleMouseEnter}
+        onMouseLeave={handleOnMouseLeave}
+      >
+        <span data-testid="trigger" {...bindHover(popupState)}>
+          {children}
+        </span>
+      </span>
+      {isHovered && (
+        <HoverPopover
+          PaperProps={{
+            className: classes.popoverPaper,
+          }}
+          {...bindPopover(popupState)}
+          anchorOrigin={{
+            vertical: 'bottom',
+            horizontal: 'center',
+          }}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'center',
+          }}
+        >
+          <>
+            {loading && <Progress />}
```

Comment:
note in the story how the progress looks (stacked at the top with no padding), it should probably be inside a card content as well

Reply:
done.